### DISCUSSION TITLE
Lesson 5: Error in running: brownie run scripts/deploy.py with a blockheight warning

### DISCUSSION THREAD

**Discussion by rishigarg256** [smartcontractkit/full-blockchain-solidity-course-py#1175] [Q&A] [answered]
getting this error
Brownie v1.18.1 - Python development framework for Ethereum

BrownieSimpleStorageProject is the active project.
C:\anaconda\lib\site-packages\brownie\network\main.py:44: BrownieEnvironmentWarning: Development network has a block height of 2
  warnings.warn(
Attached to local RPC client listening at '127.0.0.1:8545'...

Running 'scripts\deploy.py::main'...
Hello!

below is the code 
```# type brownie in the terminal to familarize with the different commands brownie has
# to deploy a script, we do brownie run
# to run this script, we do brownie run scripts/deploy.py
# brownie will spin up a local ganache chain by default to deploy to
def main():  # function main
    print("Hello!")
```
instead of giving this output
![image](https://user-images.githubusercontent.com/49986448/155920293-735100c3-b401-4baa-b9e9-710abefa579e.png)

**Comment by theref** [comment, marked as answer]
There isn't any issue here. Brownie has given you a warning and then executed the code as expected.

**Comment by cromewar** [comment]
Hello @rishigarg256 
This not seems like an error as  @theref suggested, however I have some recommendations for you:
1: Please do not use anaconda environments, they have problems with web3 libraries and connection with ganache.
2. install python version 3.8 or 3.9.
3. Create a new virtual env with that version and brownie with brownie there (install it with pip instead of pipx).

**Reply by n4n0b1t3** [reply]
In order to avoid a host of problems, it is good practice to have a standard workspace setup that works for others as well.
In this discussion here, I discuss [How to use a different Python version in a virtual environment to solve version compatibility issues with brownie](https://github.com/smartcontractkit/full-blockchain-solidity-course-py/discussions/1109). This article also contains a walk-through to set up brownie and ganache and what phyton version I am using.

**Reply by rishigarg256** [reply]
whats the difference between a virtual environment and the terminal i use for vs code?

**Reply by rishigarg256** [reply]
because i just use the terminal of the vs code.

**Reply by cromewar** [reply]
A virtual en is an isolated instance of python, vscode just uses the interpreter you ask it to use, you must select it on the bottom left corner in vscode.

**Comment by n4n0b1t3** [comment]
OK, let's unpack the question. When using Visual Studio Code on Windows the default command-line shell / integrated terminal will be [PowerShell](https://code.visualstudio.com/docs/editor/integrated-terminal). There is no difference if you are using PowerShell from the Windows menu or the integrated terminal which comes with VSC.
 
**Global vs Virtual Environment**
Now the next statement is true for both your integrated terminal or on a separate PowerShell window, you should treat them actually like one thing.
When you are using `pip install something` this python library "something" will be installed globally. This will become messy very soon. And it will become ugly when you need for example a different version of a library or of python. This is exactly what will happen when you want to follow the whole tutorial. At the time of writing, the newest brownie version will not install with the newest Python version. Of course, you could now just install an older version of Python on your system. But imagine you are at a point, where you have developed some very nice applications, which make use of the newest Python features. And bam! All of this won't work anymore when you downgrade your main (global) Python version.

**One very good way to solve this is a so-called virtual environment.**
It is virtual in a way, that you would need to activate it first to take effect. How to set up such an environment is described here: https://github.com/smartcontractkit/full-blockchain-solidity-course-py/discussions/1109.
